aspnet/client-side/bundling-and-minification.rst

```diff
@@ -1,10 +1,171 @@
 
-|stub-icon| Bundling and Minification
-=====================================
+Bundling and minification are two techniques you can use in ASP.NET to improve page load performance for your web application. Bundling combines multiple files into a single file. Minification performs a variety of different code optimizations to scripts and CSS, which results in smaller payloads. Used together, bundling and minification improves load time performance by reducing the number of requests to the server and reducing the size of the requested assets (such as CSS and JavaScript files).
 
-.. include:: /../common/stub-notice.txt
+This article explains the benefits of using bundling and minification, including how these features can be used with ASP.NET 5 applications. 
 
-.. _issue: https://github.com/aspnet/Docs/issues/101
\ No newline at end of file
+.. contents:: In this article:
+  :local:
+  :depth: 1
+
+Overview
+--------
+
+In ASP.NET Core apps, you bundle and minify the client-side resources during design-time using third party tools, such as :doc:`Gulp <using-gulp>` and :doc:`Grunt <using-grunt>`. By using design-time bundling and minification, the minified files are created prior to the application’s deployment. Bundling and minifying ahead of time provides the advantage of fewer moving parts and reduced server load. However, it’s important to recognize that design-time bundling and minification increases build complexity and only works with static files.
+
+Bundling and minification primarily improve the first page request load time. Once a web page has been requested, the browser caches the assets (JavaScript, CSS and images) so bundling and minification won’t provide any performance boost when requesting the same page, or pages on the same site requesting the same assets. If you don’t set the expires header correctly on your assets, and you don’t use bundling and minification, the browsers freshness heuristics will mark the assets stale after a few days and the browser will require a validation request for each asset. In this case, bundling and minification provide a performance increase even after the first page request. 
+
+Bundling
+--------
+
+Bundling is a feature that makes it easy to combine or bundle multiple files into a single file. Because bundling combines multiple files into a single file, it reduces the number of requests to the server that is required to retrieve and display a web asset, such as a web page. You can create CSS, JavaScript and other bundles. Fewer files, means fewer HTTP requests from your browser to the server or from the service providing your application. This results in improved first page load performance.
+
+Bundling can be accomplished using the `gulp-concat <https://www.npmjs.com/package/gulp-concat>`__ task, which is installed via the Node Package Manager (`npm <https://www.npmjs.com/>`__). Add the ``gulp-concat`` package to the ``devDependencies`` section of your ``packages.json`` file:
```

Comment:
Perhaps a quick sentence or two defining npm? Many legacy .NET devs and non JS devs may not know what it is.